resources/lib/gui.py

```diff
@@ -992,6 +983,7 @@ def onClick(self, controlID):
         if controlID == 307:
             # Change Action
             log( "Change action (307)" )
+            currentWindow = xbmcgui.Window( xbmcgui.getCurrentWindowDialogId() )
```

Comment:
I always assign self.window_id in onInit() already since for most WindowXML instances it is used pretty often in all kind of methods.